src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,33 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(health, name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!animal.hidden && animal instanceof Herbivore) {
+      animal.health -= 50;
+
+      Animal.alive = Animal.alive
+        .filter(value => value.health > 0);
+    }
```

Comment:
Would be better to filter alive array only when the animal heath `<= 0`
    
```suggestion
    if (!animal.hidden && animal instanceof Herbivore) {
      animal.health -= 50;
    }
    
    if(alimal.heath <= 0) {
      Animal.alive = Animal.alive.filter(value => value.health > 0);
    }
```